src/telemetry/filters.ts

```diff
@@ -0,0 +1,74 @@
+import { ITelemetryItem } from '@microsoft/applicationinsights-web';
+import {
+  DEVX_API_URL,
+  GRAPH_API_SANDBOX_URL,
+  GRAPH_URL,
+} from '../app/services/graph-constants';
+import {
+  sanitizeGraphAPISandboxUrl,
+  sanitizeQueryUrl,
+} from '../app/utils/query-url-sanitization';
+
+export function filterTelemetryTypes(envelope: ITelemetryItem) {
+  const baseType = envelope.baseType || '';
+  const telemetryTypesToInclude = [
+    'EventData',
+    'MetricData',
+    'PageviewData',
+    'ExceptionData',
+    'RemoteDependencyData',
+  ];
+  return telemetryTypesToInclude.includes(baseType);
+}
+
+export function filterRemoteDependencyData(envelope: ITelemetryItem): boolean {
+  const baseData = envelope.baseData || {};
+  if (envelope.baseType !== 'RemoteDependencyData') {
+    return true;
+  }
+
+  const targetsToInclude = [GRAPH_URL, DEVX_API_URL, GRAPH_API_SANDBOX_URL];
+  const urlObject = new URL(baseData.target || '');
+  if (!targetsToInclude.includes(urlObject.origin)) {
+    return false;
+  }
+
+  const target = baseData.target || '';
+  switch (urlObject.origin) {
+    case GRAPH_URL:
+      baseData.name = sanitizeQueryUrl(target);
+      break;
+    case GRAPH_API_SANDBOX_URL:
+      baseData.name = sanitizeGraphAPISandboxUrl(target);
+    default:
+      break;
+  }
+  return true;
+}
+
+export function addCommonTelemetryItemProperties(envelope: ITelemetryItem) {
+  const telemetryItem = envelope.baseData || {};
+  telemetryItem.properties = telemetryItem.properties || {};
+
+  // Identifies the source of telemetry events
+  telemetryItem.properties.ApplicationName = 'Graph Explorer v4';
```

Comment:
We should capture the version of Graph Explorer that is emitting these events. After every deployment of Graph Explorer, there is a period where there are older versions of Graph Explorer loaded into a user's browsers and CDN caches haven't been updated with the latest bundle. We could have an old and new version emitting events at the same time. Capturing information about version and other variation identifiers could be useful in the future in case we want to do AB testing.

Reply:
I've created an [issue](https://github.com/microsoftgraph/microsoft-graph-explorer-v4/issues/905) for capturing version information